sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -384,7 +384,7 @@ private async Task PopulatePulumiVersionAsync(CancellationToken cancellationToke
 
         internal static SemVersion? ParseAndValidatePulumiVersion(SemVersion minVersion, string currentVersion, bool optOut)
         {
-            if (!SemVersion.TryParse(currentVersion, out SemVersion? version))
+            if (!SemVersion.TryParse(currentVersion, SemVersionStyles.Any, out SemVersion? version))
```

Comment:
All these `SemVersion`-related changes were made because `SemVersion.Parse(string)` and `SemVersion.Parse(string, bool)` was made obsolete. The equivalent is passing `SemVersionStyles.Any`, which is the most permissive format.